src/webgpu/api/validation/createBindGroupLayout.spec.ts

```diff
@@ -26,27 +26,35 @@ function clone<T extends GPUBindGroupLayoutDescriptor>(descriptor: T): T {
 
 export const g = makeTestGroup(ValidationTest);
 
-g.test('some_binding_index_was_specified_more_than_once').fn(async t => {
-  const goodDescriptor = {
-    entries: [
-      { binding: 0, visibility: GPUShaderStage.COMPUTE, type: 'storage-buffer' as const },
-      { binding: 1, visibility: GPUShaderStage.COMPUTE, type: 'storage-buffer' as const },
-    ],
-  };
+g.test('some_binding_index_was_specified_more_than_once')
+  .desc('Test that uniqueness of binding numbers across entries is enforced.')
+  .fn(async t => {
+    const goodDescriptor = {
```

Comment:
Could I ask that we refactor this and a few others in this file to use a subcases-based control case instead of the goodDescriptor badDescriptor thing? At least, the ones for which the refactor is easy. (And then possibly delete the `clone` function if it's unused)

This could be a follow-up PR; up to you.

Reply:
Refactored a few of these. The last couple of tests are a bit trickier, so I'll try to do that in a followup.